### DISCUSSION TITLE
Getting unconverted values in entity constructor with MikroOrm v6

### DISCUSSION THREAD

**Discussion by mjroeleveld** [mikro-orm/mikro-orm#5117] [Q&A]
I switched to MikroOrm v6 and most is working fine but I have started receiving unconverted types for properties using a custom `type` (previously `customType`) in constructors. I'm using `forceEntityConstructor: true`. I have set up a minimal reproduction but it seems to work in a minimal setup. I'm very confused, any hints would be greatly appreciated.

**Comment by B4nan** [comment]
Hard to guess without seeing any code... Try to start with your broken app state and simplify it till it starts to work.

> forceEntityConstructor: true

May I ask why? So far I've seen very few people who used this flag when they had to, most people completely misunderstand it and blindly enable it for no reason... It affects only managed entities, and is needed only for one very specific use case (using native private fields).

**Comment by mjroeleveld** [comment]
@B4nan I made a reconstruction for the reason why: non-native private fields won't work without it.

```typescript
import { Entity, MikroORM, PrimaryKey, Property, Type } from '@mikro-orm/core';
import { SqliteDriver } from '@mikro-orm/sqlite';

@Entity()
class TestEntity {
  private test = [];

  @PrimaryKey()
  public readonly id: number;

  public testProp() {
    return typeof this.test === 'object';
  }

  constructor(props: { id: number }) {
    this.id = props.id;
  }
}

describe('Q&A', () => {
  let orm: MikroORM;

  beforeAll(async () => {
    orm = await MikroORM.init({
      dbName: 'mirko-orm-test',
      driver: SqliteDriver,
      entities: [TestEntity],
      entitiesTs: undefined,
    });
    await orm.schema.dropSchema();
    await orm.schema.createSchema();
  });

  afterAll(async () => {
    await orm.close(true);
  });

  test(`Q&A`, async () => {
    const em = orm.em.fork();
    const testEntity = new TestEntity({
      id: 1,
    });
    await em.persistAndFlush(testEntity);
    em.clear();
    const retrieved = await em.findOneOrFail(TestEntity, {
      id: 1
    });
    expect(retrieved.testProp()).toStrictEqual(true); // This is false
  });
});

**Reply by mjroeleveld** [reply]
I need private fields because of business logic that I'm performing on the entity. Without the flag, I cannot use them.

**Reply by B4nan** [reply]
You can use an `@OnLoad` hook too, but yeah, it's fine to use the option for this use case, as long as you understand what it does.
